aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -20,14 +19,203 @@
 #include <c10/util/irange.h>
 #include <c10/core/QScheme.h>
 #include <c10/util/SmallVector.h>
+#include <ATen/native/IndexKernel.h>
 
 #ifdef __OBJC__
 #include <MetalPerformanceShaders/MetalPerformanceShaders.h>
 #endif
 
+extern const char * index_select_kernel_mps;
+
 namespace at {
 namespace native {
 
+static
+bool selectIndexFunctionName(ScalarType scalar_type, std::string& indexFunctionName) {
```

Comment:
I can imagine this to be useful for other kernels as well, can we move this to OperationUtils.mm?